pros/src/async_runtime/executor.rs

```diff
@@ -0,0 +1,121 @@
+use core::{
+    cell::RefCell,
+    future::Future,
+    pin::Pin,
+    task::{Context, Waker},
+};
+
+use alloc::{boxed::Box, collections::VecDeque, rc::Rc, task::Wake};
+
+use crate::task_local;
+
+use super::{reactor::Reactor, JoinHandle};
+
+task_local! {
+    pub(crate) static EXECUTOR: Rc<Executor> = Rc::new(Executor::new())
```

Comment:
Does this mean async can't be used between multiple threads?

Reply:
I'm not entirely sure what you mean by this. If it helps, every freertos task has its own executor because the executor is single threaded. If tasks shared the same executor a future being polled in one task would block a future being polled in a completely different one, which could dramatically slow things down in certain circumstances.

Reply:
What I meant is that if I were to make a future and send it to another thread, could I use it there? I'm guessing yes but I'm not positive.

Reply:
If you register a future in the executor there is no way to send it. The executor owns futures. This unsafe impl is on the TaskWaker struct which only is ever created in an executor. That is why it is fine to unsafe impl.